FiveRockingFingers/FRF.Core/Services/ArtifactsService.cs

```
@@ -193,14 +199,19 @@ public async Task<ServiceResponse<IList<ArtifactsRelation>>> SetRelationAsync(IL
             return new ServiceResponse<IList<ArtifactsRelation>>(_mapper.Map<IList<ArtifactsRelation>>(resultArtifactRelations));
         }
 
-        public async Task<IList<ArtifactsRelation>> GetRelationsAsync(int artifactId)
+        public async Task<ServiceResponse<IList<ArtifactsRelation>>> GetAllRelationsOfAnArtifactAsync(int artifactId)
         {
-            var artifactsRelations = await _dataContext.ArtifactsRelation
-                .Include(ar => ar.Artifact1)
+            var result = await _dataContext.ArtifactsRelation.Include(ar => ar.Artifact1)
                 .Include(ar => ar.Artifact2)
-                .Where(ar => ar.Artifact1Id == artifactId || ar.Artifact2Id == artifactId).ToListAsync();
-            var resultArtifactRelations = _mapper.Map<List<ArtifactsRelation>>(artifactsRelations);
-            return resultArtifactRelations;
+                .Where(ar => ar.Artifact1Id == artifactId || ar.Artifact2Id == artifactId)
+                .ToListAsync();
+
+            if (!result.Any())
+                return new ServiceResponse<IList<ArtifactsRelation>>(new Error(ErrorCodes.RelationNotExist,
+                    $"There is no relations for the artifact with Id = {artifactId}"));
```

I don't know if this should return an error, maybe a successful response with an empty list is better.

---

Resolved here #78.